src/modules/common/mentors.service.ts

```diff
@@ -15,13 +16,25 @@ export class MentorsService {
     private readonly applicationModel: Model<Application>,
   ) {}
 
+  /**
+   * Finds a mentor by ID
+   * @param _id
+   */
+  async findById(_id: string): Promise<User> {
+    if (isObjectId(_id)) {
+      return await this.userModel.findOne({ _id, roles: Role.MENTOR }).exec();
```

Comment:
Not sure I understand the use of `async` `await` here.
```suggestion
  findById(_id: string): Promise<User> {
    if (isObjectId(_id)) {
      return this.userModel.findOne({ _id, roles: Role.MENTOR }).exec();
```

Reply:
There's no need for that await in there, will remove it!